zerver/management/commands/makemessages.py

```diff
@@ -269,6 +276,15 @@ def get_new_strings(
             else:
                 new_strings[k] = old_strings.get(k, "")
 
+        if self.frontend_keep_stream_strings:
+            for old in old_strings:
+                # If the existing string has "stream" in it, keep it.
+                if "stream" in old.lower() and old not in new_strings:
+                    if locale == "en":
+                        new_strings[old] = old
+                    else:
+                        new_strings[old] = ""
```

Comment:
I think what you want is for the `en` frontend strings output to not have an entry (that's what we upload to Transifex):
```
$ cat tools/i18n/push-translations 
#!/usr/bin/env bash

# See https://zulip.readthedocs.io/en/latest/translating/internationalization.html
# for background on this subsystem.

set -e
set -x

./manage.py makemessages --locale=en
tx push -s -r zulip.djangopo,zulip.translationsjson "$@"
```

But for the other languages to use whatever is cached locally on disk, if there is something.

```
$ cat tools/i18n/sync-translations 
#!/usr/bin/env bash

# See https://zulip.readthedocs.io/en/latest/translating/internationalization.html
# for background on this subsystem.

set -e
set -x

./manage.py makemessages --all
tx pull -a -f --mode=translator --minimum-perc=5 "$@"
# For readability, we prefer UTF-8, not ascii, in these JSON files.
find ./locale \
    -regextype sed \
    -regex '^\./locale/.*/\(mobile\|translations\).json$' \
    -exec ./tools/i18n/unescape-contents {} \;

# Normalize to Unicode Normalization Form C.
files="$(find locale -type f -name '*.json' -o -name '*.po')"
mapfile -t files <<<"$files"
for file in "${files[@]}"; do
    uconv -x any-nfc "$file" | sponge -- "$file"
done

./manage.py compilemessages --ignore='*'
./tools/i18n/process-mobile-i18n

```

I'm not sure what `tx pull` will do if there are strings in the local files that are not present remotely, but I think it might just overwrite the local translation file entirely. If so, I don't totally know how to make this strategy work without forking `tx`, which seems like a pain... 

I suppose a credible option would be to just put a `legacy_stream_translations.json` in each directory that's generated by a one-off script that just extracts all the entries from the existing `translations.json` (or probably better, a `tx pull` download run using the commit just before we started changing strings from stream=>channel to get any Translator updates in the relevant weeks) that contain the word "stream", and have `makemessages` merge those into the `translations.json` file for strings downloaded from post-upload-rename Transifex, if there's no translation available in the synced copy of `translations.json` ... that seems like a pretty clean API, and wouldn't require complex changes in any other part of the pipeline. (I guess we'd miss the `django.po`, but that seems OK -- those are mostly error messages and much lower visibility)